frontend/javascripts/oxalis/model/sagas/quick_select_saga.ts

```diff
@@ -0,0 +1,596 @@
+import {
+  CancelQuickSelectAction,
+  ComputeQuickSelectForRectAction,
+  ConfirmQuickSelectAction,
+  FineTuneQuickSelectAction,
+  finishAnnotationStrokeAction,
+  registerLabelPointAction,
+} from "oxalis/model/actions/volumetracing_actions";
+import BoundingBox from "oxalis/model/bucket_data_handling/bounding_box";
+import api from "oxalis/api/internal_api";
+import ndarray from "ndarray";
+import morphology from "ball-morphology";
+import Toast from "libs/toast";
+import { DatasetLayerConfiguration, OxalisState, VolumeTracing } from "oxalis/store";
+import { RectangleGeometry } from "oxalis/geometries/helper_geometries";
+import { clamp, take2 } from "libs/utils";
+import { APIDataLayer } from "types/api_flow_types";
+import { copyNdArray } from "./volume/volume_interpolation_saga";
+import { createVolumeLayer, labelWithVoxelBuffer2D } from "./volume/helpers";
+import { EnterAction, EscapeAction, setIsQuickSelectActiveAction } from "../actions/ui_actions";
+import {
+  getEnabledColorLayers,
+  getLayerBoundingBox,
+  getResolutionInfo,
+} from "../accessors/dataset_accessor";
+import Dimensions from "../dimensions";
+import { getRequestLogZoomStep } from "../accessors/flycam_accessor";
+import { updateUserSettingAction } from "../actions/settings_actions";
+
+// How large should the center rectangle be.
+// Used to determine the mean intensity.
+const CENTER_RECT_SIZE_PERCENTAGE = 1 / 10;
+
+export default function* listenToMinCut(): Saga<void> {
+  yield* takeEvery(
+    "COMPUTE_QUICK_SELECT_FOR_RECT",
+    function* guard(action: ComputeQuickSelectForRectAction) {
+      try {
+        yield* put(setIsQuickSelectActiveAction(true));
+        yield* call(performQuickSelect, action);
+      } catch (ex) {
+        Toast.error((ex as Error).toString());
+        ErrorHandling.notify(ex as Error);
+        console.error(ex);
+      } finally {
+        yield* put(setIsQuickSelectActiveAction(false));
+      }
+    },
+  );
+}
+
+function* performQuickSelect(action: ComputeQuickSelectForRectAction): Saga<void> {
+  const activeViewport = yield* select(
+    (state: OxalisState) => state.viewModeData.plane.activeViewport,
+  );
+  const { rectangleGeometry } = action;
+  if (activeViewport === "TDView") {
+    // Can happen when the user ends the drag action in the 3D viewport
+    console.warn("Ignoring quick select when mouse is in 3D viewport");
+    rectangleGeometry.setCoordinates([0, 0, 0], [0, 0, 0]);
+    return;
+  }
+  const [firstDim, secondDim, thirdDim] = Dimensions.getIndices(activeViewport);
+  const quickSelectConfig = yield* select((state) => state.userConfiguration.quickSelect);
+
+  const colorLayers = yield* select((state: OxalisState) =>
+    getEnabledColorLayers(state.dataset, state.datasetConfiguration),
+  );
+  if (colorLayers.length === 0) {
+    Toast.warning("No color layer available to use for quickSelect feature");
+    return;
+  }
+
+  const colorLayer = colorLayers[0];
```

Comment:
You are right. I added a warning toast (will only be shown once) in case there are multiple color layers. I the long run, we probably want to make this configurable, but I'd wait for real user feedback first.